src/bin/activate.rs

```diff
@@ -348,70 +352,76 @@ pub async fn activate(
     temp_path: String,
     confirm_timeout: u16,
     magic_rollback: bool,
+    dry_activate: bool,
 ) -> Result<(), ActivateError> {
-    info!("Activating profile");
-
-    let nix_env_set_exit_status = Command::new("nix-env")
-        .arg("-p")
-        .arg(&profile_path)
-        .arg("--set")
-        .arg(&closure)
-        .status()
-        .await
-        .map_err(ActivateError::SetProfileError)?;
-
-    match nix_env_set_exit_status.code() {
-        Some(0) => (),
-        a => {
-            if auto_rollback {
-                deactivate(&profile_path).await?;
+    if !dry_activate {
+        info!("Activating profile");
+        let nix_env_set_exit_status = Command::new("nix-env")
+            .arg("-p")
+            .arg(&profile_path)
+            .arg("--set")
+            .arg(&closure)
+            .status()
+            .await
+            .map_err(ActivateError::SetProfileError)?;
+        match nix_env_set_exit_status.code() {
+            Some(0) => (),
+            a => {
+                if auto_rollback && !dry_activate {
+                    deactivate(&profile_path).await?;
+                }
+                return Err(ActivateError::SetProfileExitError(a));
             }
-            return Err(ActivateError::SetProfileExitError(a));
-        }
-    };
+        };
+    }
 
     debug!("Running activation script");
 
     let activate_status = match Command::new(format!("{}/deploy-rs-activate", profile_path))
-        .env("PROFILE", &profile_path)
-        .current_dir(&profile_path)
+        .env("PROFILE", &closure)
+        .env("DRY_ACTIVATE", if dry_activate { "1" } else { "0" })
+        .current_dir(&closure)
```

Comment:
I don't think `closure` should be used in place of `profile_path` unconditionally, only when `dry_activate` is true. Alternatively, we could also just add a different environment variable, i.e. `$CLOSURE`, and have that be used within the dry-activate scripts instead of `$PROFILE`